nifi-extension-bundles/nifi-jolt-bundle/nifi-jolt-processors/src/main/java/org/apache/nifi/processors/jolt/JoltTransformJSON.java

```diff
@@ -55,11 +57,31 @@
 @Tags({"json", "jolt", "transform", "shiftr", "chainr", "defaultr", "removr", "cardinality", "sort"})
 @InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
 @WritesAttribute(attribute = "mime.type", description = "Always set to application/json")
-@CapabilityDescription("Applies a list of Jolt specifications to the flowfile JSON payload. A new FlowFile is created "
-        + "with transformed content and is routed to the 'success' relationship. If the JSON transform "
-        + "fails, the original FlowFile is routed to the 'failure' relationship.")
+@CapabilityDescription("Applies a list of JOLT specifications to either the FlowFile JSON content or a specified FlowFile JSON attribute. "
+        + "When 'Json Source' is set to FLOW_FILE, the FlowFile content is transformed and the modified FlowFile is routed to the 'success' relationship. "
+        + "When 'Json Source' is set to ATTRIBUTE, the specified attribute's value is transformed and updated in place, with the FlowFile routed to the 'success' relationship. "
+        + "If the JSON transform fails, the original FlowFile is routed to the 'failure' relationship.")
 @RequiresInstanceClassLoading
 public class JoltTransformJSON extends AbstractJoltTransform {
+
+    public static final PropertyDescriptor JSON_SOURCE = new PropertyDescriptor.Builder()
+            .name("Json Source")
+            .description("Specifies whether the JOLT transformation is applied to FlowFile JSON content or to specified FlowFile JSON attribute.")
+            .required(true)
+            .allowableValues(SourceStrategy.class)
+            .defaultValue(SourceStrategy.FLOW_FILE)
+            .build();
+
+    public static final PropertyDescriptor JSON_SOURCE_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("Json Source Attribute")
+            .description("The FlowFile attribute containing JSON to be transformed. "
+                    + "This property is required only when 'Json Source' is set to ATTRIBUTE.")
```

Comment:
This can be specified as a multiline string, however, it is not necessary to include the second sentence, since the documentation rendering automatically describes dependent properties.